src/web/views/FlowBuilder/skills/index.jsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react'
+import { Modal, Button } from 'react-bootstrap'
+import axios from 'axios'
+import find from 'lodash/find'
+import includes from 'lodash/includes'
+import Loader from 'halogen/BounceLoader'
+
+const style = require('./style.scss')
+
+const VALID_WINDOW_SIZES = ['normal', 'large', 'small']
+
+import InjectedModuleView from '~/components/PluginInjectionSite/module'
+
+class WrappedInjectedModule extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    return nextProps.moduleProps !== this.props.moduleProps || nextProps.moduleName !== this.props.moduleName
+  }
+
+  render() {
+    return <InjectedModuleView {...this.props} />
+  }
+}
+
+export default class SkillsBuilder extends React.Component {
+  resetState = () => ({
+    moduleProps: {},
+    canSubmit: false,
+    loading: false,
+    windowSize: 'normal'
+  })
+
+  state = this.resetState()
+
+  componentDidMount() {
+    this.setState({ initialData: this.props.data })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.skillId !== this.props.skillId || nextProps.opened !== this.props.opened) {
+      this.setState({
+        ...this.resetState(),
+        moduleProps: this.buildModuleProps(nextProps.data)
```

Comment:
This method is not called for the initial props during mounting. Is it fine?

Reply:
In this case it's fine but if we wanted this component to be Pure it wouldn't. Will add it on componentDidMount as well for the sake of good practice :)